code/modules/reagents/chemistry/reagents/cat2_medicine_reagents.dm

```diff
@@ -524,7 +524,23 @@
 		to_chat(carbies, span_danger("You feel your burns and bruises healing! It stings like hell!"))
 
 	carbies.add_mood_event("painful_medicine", /datum/mood_event/painful_medicine)
-	if(HAS_TRAIT_FROM(exposed_mob, TRAIT_HUSK, BURN) && carbies.getFireLoss() < UNHUSK_DAMAGE_THRESHOLD && (carbies.reagents.get_reagent_amount(/datum/reagent/medicine/c2/synthflesh) + reac_volume >= SYNTHFLESH_UNHUSK_AMOUNT))
+
+	//don't unhusked non husked mobs
+	if (!HAS_TRAIT_FROM(exposed_mob, TRAIT_HUSK, BURN))
+		return
+
+	//don't try to unhusk mobs above burn damage threshold
+	if (carbies.getFireLoss() > UNHUSK_DAMAGE_THRESHOLD)
+		return
+
+	var/current_volume = carbies.reagents.get_reagent_amount(/datum/reagent/medicine/c2/synthflesh)
+	var/current_purity = carbies.reagents.get_reagent_purity(/datum/reagent/medicine/c2/synthflesh)
+	if (methods & TOUCH)	//touch does not apply chems to blood, we want to combine the two volumes before attempting to unhusk
+		current_purity = current_volume > 0 ? (current_volume * current_purity + reac_volume * creation_purity) / (current_volume + reac_volume) : creation_purity
+		current_volume += reac_volume
+
+	//when purity = 100%, 60u to unhusk, when purity = 60%, 100u to unhusk.
+	if(current_volume * current_purity >= SYNTHFLESH_UNHUSK_AMOUNT || current_volume >= SYNTHFLESH_UNHUSK_MAX)
```

Comment:
```suggestion
	if(current_volume >= SYNTHFLESH_UNHUSK_MAX || current_volume * current_purity >= SYNTHFLESH_UNHUSK_AMOUNT)
```
Order of conditions here slightly improves performance as in if if the volume is greater than `SYNTHFLESH_UNHUSK_MAX` then we shortcircuit & skip computing `current_volume * current_purity`